### ISSUE TITLE
OpenBLAS ARM compilation

### ISSUE THREAD

**Issue by Rtik1988** [xianyi/OpenBLAS#1805]
The instructions for cross-compilation in this [wiki page](https://github.com/xianyi/OpenBLAS/wiki/How-to-build-OpenBLAS-for-Android#building-openblas-with-fortran) result in static '.a' file.
I want to create a dynamic library (so) to package as part of an android app.

I've tried running (after libs target compilation):
```
make TARGET=ARMV7 HOSTCC=gcc CC=arm-linux-androideabi-gcc FC=arm-linux-androideabi-gfortran shared
```
But it fails with very large number of errors of that type:
```
/tmp/ccEmpsUH.o:linktest.c:function main: error: undefined reference to 'LAPACKE_xxxxxxxxx'
/tmp/ccEmpsUH.o:linktest.c:function main: error: undefined reference to 'LAPACKE_xxxxxxxxx_work'
```
Where the 'xxxxxxxx' are names of LAPACK function names.

I was able to solve this by first running:
```
make TARGET=ARMV7 HOSTCC=gcc CC=arm-linux-androideabi-gcc FC=arm-linux-androideabi-gfortran netlib
```
and then rerunning:
```
make TARGET=ARMV7 HOSTCC=gcc CC=arm-linux-androideabi-gcc FC=arm-linux-androideabi-gfortran shared
```

Does this mean that the OpenBLAS `libopenblas_armv7p-r0.3.4.dev.so` object uses netlib implementation of BLAS? Thus making it slower and sub-optimal.
Or does it only uses the LAPACKE C wrapper of LAPACK?

**Comment by martin-frbg** [comment]
`As mentioned above you can leave off the libs argument here when building 0.2.20 and later` is maybe not clear enough ? The "mentioned above" explains that it was not possible to build a shared library with old versions up to 0.2.19, so the instructions were to build only the "libs" step (static library).
The "netlib" in your shared library is the LAPACK and LAPACKE parts, which are taken from the netlib reference implementation included as lapack-netlib. BLAS is always the OpenBLAS implementation

**Comment by Rtik1988** [comment]
Thank you for the quick response.

The explanation is totally clear, I've first tried to run:
```
make TARGET=ARMV7 HOSTCC=gcc CC=arm-linux-androideabi-gcc FC=arm-linux-androideabi-gfortran
``` 

But the build of test target failed with lots of undefined reference errors such as:
```
/root/toolchain/build/../gcc/gcc-4.9/libgfortran/io/list_read.c:2390: error: undefined reference to 'strcmp'
/root/toolchain/build/../gcc/gcc-4.9/libgfortran/io/list_read.c:2390: error: undefined reference to 'strcmp'
/root/toolchain/build/../gcc/gcc-4.9/libgfortran/io/list_read.c:2969: error: undefined reference to 'strchr'
/root/toolchain/build/../gcc/gcc-4.9/libgfortran/io/list_read.c:3026: error: undefined reference to 'strchr'
/root/toolchain/build/../gcc/gcc-4.9/libgfortran/io/read.c:163: error: undefined reference to 'strtod'
/root/toolchain/build/../gcc/gcc-4.9/libgfortran/io/read.c:158: error: undefined reference to 'strtod'
/root/toolchain/build/../gcc/gcc-4.9/libgfortran/io/read.c:965: error: undefined reference to 'isalnum'
/root/toolchain/build/../gcc/gcc-4.9/libgfortran/io/read.c:1111: error: undefined reference to '__assert2'
/root/toolchain/build/../gcc/gcc-4.9/libgcc/config/arm/lib1funcs.S:1337: error: undefined reference to 'raise'
/root/toolchain/build/../gcc/gcc-4.9/libgcc/config/arm/pr-support.c:378: error: undefined reference to 'abort'
/root/toolchain/build/../gcc/gcc-4.9/libgcc/config/arm/pr-support.c:384: error: undefined reference to 'abort'
/root/toolchain/build/../gcc/gcc-4.9/libgcc/unwind-arm-common.inc:289: error: undefined reference to 'abort'
collect2: error: ld returned 1 exit status
Makefile:146: recipe for target 'zblat1' failed
make[1]: *** [zblat1] Error 1
```

It seems like it fails to find any standard functions.

**Comment by martin-frbg** [comment]
Yes that looks as if the build of the test target does not link to libc at all.

**Comment by martin-frbg** [comment]
Works for me with the (now somewhat outdated) precompiled toolchain from buffer51 as linked on the wiki page - though his r13b lacked the sysroot branch (which I copied from his r12b tarball) and I had to fix a stray include in one of the utests (ccd5945). You will probably want to add ARM_SOFTFP_ABI=1 to the `make` arguments to keep if from trying to link against the incompatible libm_hard as well. (softfp seems to be the default in this particular toolchain, but Makefile.system still needs to see the ARM_SOFTFP_ABI option to do the right thing...)
